sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample10_RunningWithDifferentStorageVersion.cs

```diff
@@ -0,0 +1,214 @@
+
+                var expectedEvents = new List<EventData>()
+                {
+                   new EventData(Encoding.UTF8.GetBytes("First Event, Single Batch")),
+                   new EventData(Encoding.UTF8.GetBytes("Second Event, Single Batch")),
+                   new EventData(Encoding.UTF8.GetBytes("Third Event, Single Batch"))
+                };
+
+                await using (var producer = new EventHubProducerClient(eventHubsConnectionString, eventHubName))
+                {
+                    using EventDataBatch eventBatch = await producer.CreateBatchAsync();
+
+                    for (int index = 0; index < expectedEvents.Count; ++index)
+                    {
+                        eventBatch.TryAdd(expectedEvents[index]);
+                    }
+
+                    await producer.SendAsync(eventBatch);
+                }
+
+                Console.WriteLine("Starting the Event Processor client...");
+                Console.WriteLine();
+
+                await processor.StartProcessingAsync();
+
+                using var cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(TimeSpan.FromSeconds(90));
+
+                while ((!cancellationSource.IsCancellationRequested) && (eventsProcessed < expectedEvents.Count))
+                {
+                    await Task.Delay(TimeSpan.FromMilliseconds(250));
+                }
+
+                // Once we arrive at this point, either cancellation was requested or we have processed all of our events.  In
+                // both cases, we'll want to shut down the processor.
+
+                Console.WriteLine();
+                Console.WriteLine("Stopping the processor...");
+
+                await processor.StopProcessingAsync();
+            }
+            finally
+            {
+                // It is encouraged that you unregister your handlers when you have finished
+                // using the Event Processor to ensure proper cleanup.  This is especially
+                // important when using lambda expressions or handlers in any form that may
+                // contain closure scopes or hold other references.
+
+                processor.ProcessEventAsync -= processEventHandler;
+                processor.ProcessErrorAsync -= processErrorHandler;
+            }
+
+            // The Event Processor client has been stopped and is not explicitly disposable; there
+            // is nothing further that we need to do for cleanup.
+
+            Console.WriteLine();
+        }
+
+        /// <summary>
+        ///   A pipeline policy to be applied to a Blob Container Client.  This policy will be applied to every
+        ///   request sent by the client, making it possible to specify the Azure Storage version they will target.
+        /// </summary>
+        ///
+        private class StorageAPIVersionPolicy : HttpPipelineSynchronousPolicy
```

Comment:
nit:  C# conventions discourage use of consecutive capitals.   We should consider `StorageApiVersionPolicy` instead.